collectors/osd.go

```diff
@@ -0,0 +1,492 @@
+package collectors
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type OsdCollector struct {
+	conn Conn
+
+	CrushWeight *prometheus.GaugeVec
```

Comment:
Add comments to these struct fields to explain what these metrics are, in the same style as: https://github.com/digitalocean/ceph_exporter/blob/master/collectors/cluster_usage.go.